test/temp

```diff
@@ -0,0 +1,67 @@
+describe('upsertWithWhere', function() {
+  var ds = getSchema();
+  var Post;
+  before('prepare "Post" model', function(done) {
+    Post = ds.define('Post', {
+      empid: { type: Number, id: true },
+      name: { type: String },
+      city: { type: String },
+    });
+    ds.automigrate('Post', done);
+  });
+  it('has an alias "patchOrCreateWithWhere"', function() {
+    StubUser.upsertWithWhere.should.equal(StubUser.patchOrCreateWithWhere);
+  });
+  it('should not allow upsert when data object is empty (where,data,options,cb)', function(done) {
+    options = undefined;
+    Post.upsertWithWhere({ name: 'John Lennon' }, {}, options,
+      function(err) {
+        should.exists(err);
+        done();
+      });
+  });
+  it('should create a new record when no matching instances are found ' +
+    '(where, data, options, cb)', function(done) {
+    Post.upsertWithWhere({ city: 'Florida' }, { name: 'Nick Carter', empid: 1 },
+      function(err, created) {
+        created.empid.should.equal(1);
+        created.name.should.equal('Nick Carter');
+        created.city.should.equal('Florida');
+        done();
+      });
+  });
+  it('should not allow upsert when multiple instances are retrieved based on the filter criteria' +
+    ' (where, data, options, cb)',
+    function(done) {
+      Post.create([
+        { empid: '2', name: 'Howie', city: 'Florida' },
+        { empid: '3', name: 'Kevin', city: 'Florida' },
+      ], function(err, instance) {
+        if (err) return done(err);
+        Post.upsertWithWhere({ city: 'Florida' }, {
+          empid: '4', name: 'Brian',
+        }, function(err, instance) {
+          should.exists(err);
+          done();
+        });
+      });
+    });
+  it('should update a record when one matching instance is found  ' +
+    '(where, data, options, cb)', function(done) {
+    options = undefined;
+    Post.create([
+      { empid: '5', name: 'Howie', city: 'Kentucky' },
+    ], function(err, instance) {
+      if (err) return done(err);
+      Post.upsertWithWhere({ city: 'Kentucky' }, {
+        name: 'Brian' }, function(err, instance) {
+        should.not.exists(err);
+        var result = instance.toObject();
+        result.empid.should.equal(instance.empid);
+        result.name.should.equal('Brian');
+        result.city.should.equal(instance.city);
+        done();
+      });
+    });
+  });
+});
```

Comment:
Please have all tests in `test/manipulation.test.js` and `test/persistence-hooks.suite.js`